channels/tsmf/client/alsa/tsmf_alsa.c

```diff
@@ -68,8 +68,7 @@ static BOOL tsmf_alsa_open(ITSMFAudioDevice *audio, const char *device)
 	TSMFAlsaAudioDevice *alsa = (TSMFAlsaAudioDevice *) audio;
 	if(!device)
 	{
-		if(!alsa->device[0])
-			strncpy(alsa->device, "default", sizeof(alsa->device));
+		strncpy(alsa->device, "default", sizeof(alsa->device));
 	}
```

Comment:
@ivan-83 aren't we now missing the case when the device is opened with no "device" set (so const char *device is NULL) but there is already a device set in ITSMFAudioDevice (the check alsa->device[0])